test/unit/lib/classes/Variables.test.js

```diff
@@ -2681,64 +2683,81 @@ module.exports = {
       });
       varProxy = new ProxyQuiredVariables(serverless);
     });
+
     afterEach(() => {
       logWarningSpy.restore();
       consoleLogStub.restore();
     });
-    it('should do nothing if variable has valid value.', () => {
-      varProxy.warnIfNotFound('self:service', 'a-valid-value');
-      expect(logWarningSpy).to.not.have.been.calledOnce;
-    });
-    describe('when variable string does not match any of syntax', () => {
-      // These situation happen when deep variable population fails
-      it('should do nothing if variable has null value.', () => {
-        varProxy.warnIfNotFound('', null);
-        expect(logWarningSpy).to.not.have.been.calledOnce;
+
+    describe('when unresolvedVariablesNotificationMode is set to "warn"', () => {
+      beforeEach(() => {
+        serverless.config.update({ unresolvedVariablesNotificationMode: 'warn' });
```

Comment:
`warn` is default, why is it needed to add this update in old tests?

Reply:
It doesn't *need* it, but the default is actually `undefined` which means the deprecation notice is shown for each of these tests - by setting it explicitly to `warn`, we avoid a lot of console noise.

Happy to remove it if you'd like :)

Reply:
Yes, let's remove it (those tests are anyway to be fully rewritten with `runServerless` variants)

Reply:
Done